### PR TITLE
William

### PR DESCRIPTION
**Pull request by TheDoplarEffect**
Working carrousel component with Movie posters.

Link not implemented yet since routing isn't working yet.
However, movie ID is stored and ready to be used.

### CODE REVIEW

**Comment by NoHara42** [comment]
What issue does this PR relate to?

**Comment by TheDoplarEffect** [comment]
Sorry, this is something i did last weekend in my free time and basically did both of these issues at once.
The carrousel of Cards and each card

**Review by NoHara42 — CHANGES_REQUESTED** [review]
See comments, do not commit `package-lock.json` files

**Review by NoHara42 — CHANGES_REQUESTED** [review]
also the pnpm-lock should not be included

File: src/components/API.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  image: string;
+  name: string;
+  role: string;
+};
+type MovieResponce = {
+  cast: Person[];
+  crew: Person[];
+} & DataResponce;
+
+class API {
```
Comment (NoHara42) [inline src/components/API.tsx:21]: I would prefer you make a custom hook for this

File: src/components/API.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import axios from "axios";
+import { useState } from "react";
+import { Link } from "react-router-dom";
+
+const APIKey = "5de0d3a9c085fde70b8c91f6f6a927f3";
```
Comment (NoHara42) [inline src/components/API.tsx:5]: this api logic should be part of a seperate PR

File: src/components/InputForm.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import axios from "axios";
```
Comment (NoHara42) [inline src/components/InputForm.tsx:1]: I'm assuming this is the searchbar, seperate PR

**Comment by NoHara42** [comment]
too many files for one PR, please limit each PR's content to the issues that it relates to

File: src/layouts/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export * from "./RootLayout";
```
Comment (NoHara42) [inline src/layouts/index.ts:1]: why do you need this

File: src/components/Movie.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import axios from "axios";
```
Comment (NoHara42) [inline src/components/Movie.tsx:1]: which issue does Movie.tsx defer to?

File: src/main.tsx [hunk 6]
```diff
@@ -1,7 +1,49 @@
...
 import ReactDOM from "react-dom/client";
+
+import { createBrowserRouter, Outlet, RouterProvider } from "react-router-dom";
+
 import "./index.css";
+import { Home, Movies, Icon3, Icon4, Login } from "./routes";
+import { RootLayout } from "./layouts";
+
+const router = createBrowserRouter([
+  {
```
Comment (NoHara42) [inline src/main.tsx:11]: Shouldn't this have been handled in the routing issue? #15

File: src/routes/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import { InputForm } from "../components/InputForm";
+
+export function Input() {
+  <InputForm></InputForm>;
```
Comment (NoHara42) [inline src/routes/Input.tsx:4]: This seems redundant

File: src/routes/Login.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { InputForm } from "../components/InputForm";
+
+export function Login() {
+  return (
+    <div>
+      <InputForm></InputForm>
```
Comment (NoHara42) [inline src/routes/Login.tsx:6]: Is this just placeholder content?

File: src/stories/API.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
```
Comment (NoHara42) [inline src/stories/API.stories.tsx:1]: having a story for something as abstract as an api is a bit odd, let's only make stories for visual elements

**Review by NoHara42 — CHANGES_REQUESTED** [review]
See comments and please extract changes to separate PRs based on the issue they concern

**Pull request closed without merging**